lib/SIL/SILFunctionType.cpp

```diff
@@ -697,7 +697,12 @@ class DestructureInputs {
 
     bool isInout = false;
     if (auto inoutType = dyn_cast<InOutType>(substType)) {
-      isInout = true;
+      // The `self` parameter of convenience initializers is considered 'inout'
+      // for semantic purposes in the AST, but at the ABI level is always
+      // passed by value in and returned out.
```

Comment:
We'd only end up here when computing the lowered initializing entry point type right? In which case this check should not matter since we only do this for designated inits.

Reply:
Convenience inits still have initializing entry points for ObjC interop. This is necessary so that the lowering for the ObjC thunk still works.